src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+
+        break;
+      case 'clear':
+        for (const property in state) {
+          delete state[property];
+        }
+
+        break;
+      default:
```

Comment:
Overall, your code seems to align well with the checklist. The only suggestion is to add a default case in your switch statement for error handling, as mentioned in point 4 in checklist

[CODE STYLE]: switch/case should always have default case for error handling.

well done